steps/rabbit/session.go

```diff
@@ -0,0 +1,334 @@
+			return false
+		}
+	}
+	return true
+}
+
+// WaitForMessageWithStandardProperties waits for a message with standard rabbitmq properties that are equal to the expected values.
+func (s *Session) WaitForMessageWithStandardProperties(ctx context.Context, timeout time.Duration, props amqp.Delivery) error {
+	return waitUpTo(timeout, func() error {
+		for _, msg := range s.Messages {
+			logrus.Debugf("Checking message: %s", msg.Body)
+			s.msg = msg
+			if err := s.ValidateMessageStandardProperties(ctx, props); err == nil {
+				s.Messages = []amqp.Delivery{msg}
+				return nil
+			}
+		}
+		return fmt.Errorf("no message received matches the standard properties")
+	})
+}
+
+// ValidateMessageStandardProperties checks if the message standard rabbitmq properties are equal the expected values.
+func (s *Session) ValidateMessageStandardProperties(ctx context.Context, props amqp.Delivery) error {
+	msg := reflect.ValueOf(s.msg)
+	expectedMsg := reflect.ValueOf(props)
+	t := expectedMsg.Type()
+	for i := 0; i < expectedMsg.NumField(); i++ {
+		if !expectedMsg.Field(i).IsZero() {
+			key := t.Field(i).Name
+			value := msg.Field(i).Interface()
+			expectedValue := expectedMsg.Field(i).Interface()
+			if value != expectedValue {
+				return fmt.Errorf("mismatch of standard rabbitmq property '%s': expected '%s', actual '%s'", key, expectedValue, value)
+			}
+		}
+	}
+	return nil
+}
+
+// ValidateMessageHeaders checks if the message rabbitmq headers are equal the expected values.
+func (s *Session) ValidateMessageHeaders(ctx context.Context, headers map[string]interface{}) error {
+	h := s.msg.Headers
+	for key, expectedValue := range headers {
+		value, found := h[key]
+		if !found {
+			return fmt.Errorf("missing rabbitmq message header '%s'", key)
+		}
+		if value != expectedValue {
+			return fmt.Errorf("mismatch of standard rabbitmq property '%s': expected '%s', actual '%s'", key, expectedValue, value)
+		}
+	}
+	return nil
+}
+
+// ValidateMessageTextBody checks if the message text body is equal the expected value.
+func (s *Session) ValidateMessageTextBody(ctx context.Context, expectedMsg string) error {
+	msg := string(s.msg.Body)
+	if msg != expectedMsg {
+		return fmt.Errorf("mismatch of message text: expected '%s', actual '%s'", expectedMsg, msg)
+	}
+	return nil
+}
+
+// ValidateMessageJSONBody checks if the message json body properties are equal the expected values.
+func (s *Session) ValidateMessageJSONBody(ctx context.Context, props map[string]interface{}) error {
+	m := golium.NewMapFromJSONBytes([]byte(s.msg.Body))
+	for key, expectedValue := range props {
+		value := m.Get(key)
+		if value != expectedValue {
+			return fmt.Errorf("mismatch of json property '%s': expected '%s', actual '%s'", key, expectedValue, value)
+		}
+	}
+	return nil
+}
+
+func waitUpTo(timeout time.Duration, f func() error) error {
+	end := time.Now().Add(timeout)
+	var err error
+	err = fmt.Errorf("time has expired without message")
```

Comment:
`err := fmt.Errorf("time has expired without message")`

definition is unnecessary